### PR TITLE
Fix and migrate unit tests to Kotlin + update CI scripts

### PR DESCRIPTION
**Pull request by prudrabhat**
## Description
This PR fixes broken tests and migrates them to Kotlin.  Additionally, updates the CI scripts to be inline with 3.x migration guide.

- [Add tests for non-ui components modified with 3.x pin flow](https://github.com/adobe/aepsdk-assurance-android/commit/b17775103d47da51d494e198f482e0c41e6de39e) : Migrate the non-ui tests to Kotlin and improve coverage. 
- [Update release scripts and workflows](https://github.com/adobe/aepsdk-assurance-android/commit/33f2c8d9b395f85702744585bc8f4da0e286070d) 
    - Update Makefile with standardized commands
    - Update config.yml with jdk17 support and test reporting
    - Update release workflows to use jdk17
    - Add support for workflow based version update
    - Change release workflows to be trggered manually
    - Update jitpack to use jdk17

`Note that the aep plugin currently does not include configuration to cover tests written with Robolectric. This causes the coverage to show lower than it is. Test metrics will be accurate once the plugin is updated`

## Motivation and Context
Fixe broken tests and migrates them to Kotlin.  Update the CI scripts to be inline with 3.x migration guide.

## How Has This Been Tested?
 - Verify that the worklow steps succeed locally.
 - Verify coverage
 -

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: .github/workflows/maven-release.yml [hunk 1]
```diff
@@ -1,34 +1,63 @@
...
+        description: 'Create tag? ("no" to skip)'
+        required: true
+        default: 'yes'
+
 jobs:
   publish:
     runs-on: ubuntu-latest
+
     steps:
-      - uses: actions/checkout@v2
```
Comment (praveek) [inline .github/workflows/maven-release.yml:10]: Can you add a similar step to verify the version numbers before release? 
https://github.com/adobe/aepsdk-userprofile-android/blob/dev-v3.0.0/.github/workflows/maven-release.yml#L32

File: code/assurance/build.gradle.kts [hunk 2]
```diff
@@ -42,5 +42,5 @@ dependencies {
     testImplementation("org.mockito:mockito-inline:4.5.1")
     testImplementation("net.sf.kxml:kxml2:2.3.0@jar")
     testImplementation("org.json:json:20171018")
-    testImplementation("org.robolectric:robolectric:4.2")
+    testImplementation("org.robolectric:robolectric:4.7")
```
[the remark this answers is not in the document]
Comment (praveek) [reply code/assurance/build.gradle.kts:45]: I don't think any other [extensions](https://github.com/search?q=org%3Aadobe%20Robolectric&type=code) use Roboelectric. We can update it in aep-plugin.

File: jitpack.yml [hunk 3]
```diff
@@ -1,5 +1,5 @@
 jdk:
-  - openjdk11
+  - openjdk17
 install:
    - echo "Attempting to publish to jitpack"
-   - make ci-publish-jitpack
+   - make core-publish-maven-local-jitpack
```
Comment (praveek) [inline jitpack.yml:5]: assurance

File: scripts/version.sh [hunk 4]
```diff
@@ -0,0 +1,131 @@
+#!/bin/bash
```
Comment (yangyansong-adbe) [inline scripts/version.sh:1]: Let's add a license header here.

File: jitpack.yml (hunk 3 above)
Comment (prudrabhat) [reply jitpack.yml]: This should have been `ci-publish-maven-local-jitpack`. Corrected it accordingly.

**Review by praveek — APPROVED** [review]

File: scripts/version.sh (hunk 4 above)
Comment (prudrabhat) [reply scripts/version.sh:1]: The current version of aep-plugin used by Assurance library does not yet have capability to add license to non-code files. I will update the plugin and apply licenses to all other files in a follow-up pr.

**Pull request merged**